modules/core/src/passes/pick-layers-pass.ts

```diff
@@ -135,7 +121,22 @@ export default class PickLayersPass extends LayersPass {
   }
 
   protected getLayerParameters(layer: Layer, layerIndex: number, viewport: Viewport): any {
-    const pickParameters = {...layer.props.parameters};
+    const {x, y, width, height} = viewport;
+    const pickParameters = {
+      ...layer.props.parameters,
+      scissorTest: true,
+      scissor: [x, y, width, height],
```

Comment:
For both perf and WebGPU readiness it's probably appropriate to align `LayersPassRenderOptions` with luma's `RenderPassProps`/`RenderPassParameters` and set scissor/depth parameters per-pass instead of per-layer.

Reply:
I've moved the scissor test to be performed on the pass, but I'm not sure about depth. It is common for layers to override the depth handling via parameters

Reply:
The picking pass only resets depth to avoid dirty context state handover (in v8 deck.props.parameters is not set before picking buffer redraw). They can be overwritten by per-layer settings. In this case we should reset the depth parameters in render pass creation. In the other cases depth parameters should override user props.